pkg/utils/test_util.go

```diff
@@ -7,8 +7,13 @@ package utils
 
```

Comment:
we need to create a separate package for the test utils, not just putting into a separate file.
Otherwise, callers may accidently use the test util by calling util package.

The package was not introduced by this PR, and could address it separately.

Reply:
Will address this and add E2E in a separate PR